src/main/java/org/opensearch/neuralsearch/processor/chunker/Chunker.java

```diff
@@ -40,7 +40,7 @@ public abstract class Chunker implements Validator, Parser {
      * @param parameters a map containing non-runtime parameters for chunking algorithms
      */
     @Override
-    public void parse(Map<String, Object> parameters) {
+    public void parse(Map<String, Object> parameters) throws IllegalArgumentException {
```

Comment:
Since `IllegalArgumentException` is a runtime exception and not a checked exception, we do not require to declare throws explicitly.  Since this method is `public` we will just let them know that method may fail with an exception. For private methods we can skip adding the exception. Nice to have picks.